fem/linearform_ext.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "linearform.hpp"
+#include "../general/forall.hpp"
+
+namespace mfem
+{
+
+LinearFormExtension::LinearFormExtension(LinearForm *lf): lf(lf) { Update(); }
+
+void LinearFormExtension::Assemble()
+{
+   const FiniteElementSpace &fes = *lf->FESpace();
+
+   MFEM_VERIFY(lf->Size() == fes.GetVSize(), "LinearForm size does not "
```

Comment:
Should we just call `MFEM_VERIFY(LinearIntegratorsCanUseDevice(this), "")`

The function `LinearIntegratorsCanUseDevice` would have to be declared in a header somewhere, but that might be clearer/avoid duplication.

Reply:
Reuse `MFEM_VERIFY(lf->SupportsDevice(), "Not supported.");`